app/main.py

```diff
@@ -1,25 +1,78 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.protected_name = f"_{name}"
+
+    def __get__(self, instance: object, owner: type) -> IntegerRange:
+        if instance is None:
```

Comment:
```suggestion
        if not instance:
```

You are returning `int` type here, not `IntegerRange`.